catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswEndpoint.java

```diff
@@ -686,9 +695,42 @@ private int deleteRecords(DeleteAction deleteAction)
         .toArray(String[]::new);
   }
 
+  private InsertAction transformInsertAction(InsertAction insertAction, String typeName) {
+    Optional<CswActionTransformer> op = cswActionTransformerProvider.getTransformer(typeName);
+    if (op.isPresent()) {
+      CswActionTransformer tr = op.get();
+      return tr.transform(insertAction);
+    } else {
+      return insertAction;
+    }
+  }
+
+  private DeleteAction transformDeleteAction(DeleteAction deleteAction, String typeName) {
+    Optional<CswActionTransformer> op = cswActionTransformerProvider.getTransformer(typeName);
+    if (op.isPresent()) {
+      CswActionTransformer tr = op.get();
+      return tr.transform(deleteAction);
+    } else {
+      return deleteAction;
+    }
+  }
+
+  private UpdateAction transformUpdateAction(UpdateAction updateAction, String typeName) {
+    Optional<CswActionTransformer> op = cswActionTransformerProvider.getTransformer(typeName);
+    if (op.isPresent()) {
```

Comment:
Seems like you could do `return op.map(tr -> tr.transform(updateAction).orElse(updateAction);`

Reply:
✏️